optuna/storages/_rdb/storage.py

```diff
@@ -81,6 +84,25 @@ def _create_scoped_session(
             )
         else:
             raise
+    except sqlalchemy_exc.OperationalError as e:
+        session.rollback()
+
+        # The following keystrings depend on the backend DB.
+        # If another issue about another DB is raised, the keystrings will be updated.
+        error_keystrings = ("1205", "Lock wait timeout exceeded; try restarting transaction")
+
+        if ignore_timeout_error and all([s in str(e.orig) for s in error_keystrings]):
+            _logger.debug(
+                "Ignoring {}. This happens due to a timeout. No exception is propagated here"
+                "because expecting a retry to be made by the calling function.".format(repr(e))
```

Comment:
```suggestion
                f"Ignoring {repr(e)}. This happens due to a timeout. No exception is propagated "
                "here because expecting a retry to be made by the calling function."
```